spring-kafka/src/main/java/org/springframework/kafka/support/serializer/ErrorHandlingDeserializer2.java

```diff
@@ -185,9 +199,27 @@ public T deserialize(String topic, Headers headers, byte[] data) {
 		}
 		catch (Exception e) {
 			deserializationException(headers, data, e);
-			return this.failedDeserializationFunction != null
-					? this.failedDeserializationFunction.apply(data, headers)
-					: null;
+			return recoverFromSupplier(topic, headers, data, e);
+		}
+	}
+
+	private T recoverFromSupplier(String topic, Headers headers, byte[] data, Exception exception) {
+		if (this.failedDeserializationFunction != null) {
+			FailedDeserializationInfo t = new FailedDeserializationInfo(topic, headers, data, this.isForKey, exception);
```

Comment:
The `t` name doesn't bring too much value.
Would be better to have really self explanatory names.